### PR TITLE
Update plugin names and deprecations for scheduler config

### PR DESCRIPTION
**Pull request by alculquicondor**
Ref kubernetes/kubernetes#95446

Part of kubernetes/enhancements#785

http://git.k8s.io/enhancements/keps/sig-scheduling/785-scheduler-component-config-api

/sig scheduling

### CODE REVIEW

**Comment by alculquicondor** [comment]
/assign @Huang-Wei 
cc @adtac

**Comment by PI-Victor** [comment]
could you please link the kep for this, i'm having trouble understanding where this is?

**Comment by alculquicondor** [comment]
Updated

**Comment by PI-Victor** [comment]
need to clarify some things

/hold

File: content/en/docs/reference/scheduling/config.md [hunk 1]
```diff
@@ -199,6 +202,9 @@ that are not enabled by default:
   configured labels. This plugin also favors spreading the Pods belonging to a
   Service across nodes.
   Extension points: `PreFilter`, `Filter`, `Score`.
+- `NodePreferAvoidPods`: Prioritizes nodes according to the node annotation
+  `scheduler.alpha.kubernetes.io/preferAvoidPods`.
```
Comment (sftim) [inline content/en/docs/reference/scheduling/config.md:206]: This annotation should also be listed in https://kubernetes.io/docs/reference/labels-annotations-taints/

File: content/en/docs/reference/scheduling/config.md [hunk 2]
```diff
@@ -182,15 +173,27 @@ extension points:
 You can also enable the following plugins, through the component config APIs,
 that are not enabled by default:
 
+- `SelectorSpread`: Favors spreading across nodes for Pods that belong to
+  {{< glossary_tooltip text="Services" term_id="service" >}},
+  {{< glossary_tooltip text="ReplicaSets" term_id="replica-set" >}} and
+  {{< glossary_tooltip text="StatefulSets" term_id="statefulset" >}}.
+  Extension points: `PreScore`, `Score`.
+- `CinderLimits`: Checks that OpenStack Cinder volume limits can be satisfied
```
Comment (sftim) [inline content/en/docs/reference/scheduling/config.md:181]: Optional extra tweak
```suggestion
- `CinderLimits`: Checks that [OpenStack Cinder](https://docs.openstack.org/cinder/) volume limits can be satisfied
```

**Review by sftim — DISMISSED** [review]

File: content/en/docs/reference/scheduling/config.md [hunk 3]
```diff
@@ -26,7 +26,7 @@ struct.
 A minimal configuration looks as follows:
 
 ```yaml
-apiVersion: kubescheduler.config.k8s.io/v1beta1
+apiVersion: kubescheduler.config.k8s.io/v1beta2
```
Comment (Huang-Wei) [inline content/en/docs/reference/scheduling/config.md:29]: Replace v1beta1 with v1beta2 in L23?

BTW: @sftim as it's now v1beta2, with slightly different API fields, should we also update L7 to v1.22?

File: content/en/docs/reference/scheduling/config.md [hunk 4]
```diff
@@ -58,6 +58,9 @@ extension points:
    Policy and are used to filter out nodes that can not run the Pod. Filters
    are called in the configured order. A pod is marked as unschedulable if no
    nodes pass all the filters.
+1. `PostFilter`: These plugins are called in their configured order when no
+   feasible nodes were found for the pod. If any postFilter plugin marks a node
```
Comment (Huang-Wei) [inline content/en/docs/reference/scheduling/config.md:62]:
```suggestion
   feasible nodes were found for the pod. If any postFilter plugin marks a Pod
```

File: content/en/docs/reference/scheduling/config.md [hunk 5]
```diff
@@ -58,6 +58,9 @@ extension points:
...
+   as `Schedulable`, the remaining plugins are not called.
```
Comment (Huang-Wei) [inline content/en/docs/reference/scheduling/config.md:63]: Prefer not to emphasize/backquote "Schedulable" as it's just an internal state:

```suggestion
   schedulable, the remaining plugins are not called.
```

File: content/en/docs/reference/scheduling/config.md [hunk 6]
```diff
@@ -199,6 +202,9 @@ that are not enabled by default:
   configured labels. This plugin also favors spreading the Pods belonging to a
   Service across nodes.
   Extension points: `PreFilter`, `Filter`, `Score`.
+- `NodePreferAvoidPods`: Prioritizes nodes according to the node annotation
+  `scheduler.alpha.kubernetes.io/preferAvoidPods`.
+  Extension points: `Score`.
```
Comment (Huang-Wei) [inline content/en/docs/reference/scheduling/config.md:208]: (not related with this line)

Update L261 to link to v1beta2.

File: content/en/docs/reference/scheduling/config.md (hunk 3 above)
Comment (alculquicondor) [reply content/en/docs/reference/scheduling/config.md:29]: Better listed the options.

I don't think we should update the min beta version.

File: content/en/docs/reference/scheduling/config.md (hunk 4 above)
Comment (alculquicondor) [reply content/en/docs/reference/scheduling/config.md:62]: the Pod.

File: content/en/docs/reference/scheduling/config.md (hunk 5 above)
Comment (alculquicondor) [reply content/en/docs/reference/scheduling/config.md:63]: Done

File: content/en/docs/reference/scheduling/config.md (hunk 2 above)
Comment (alculquicondor) [reply content/en/docs/reference/scheduling/config.md:181]: Done

File: content/en/docs/reference/scheduling/config.md (hunk 1 above)
Comment (alculquicondor) [reply content/en/docs/reference/scheduling/config.md:206]: Done

File: content/en/docs/reference/scheduling/config.md (hunk 6 above)
Comment (alculquicondor) [reply content/en/docs/reference/scheduling/config.md:208]: Done

File: content/en/docs/reference/scheduling/config.md (hunk 5 above)
Comment (sftim) [reply content/en/docs/reference/scheduling/config.md]: Maybe put _schedulable_ in italics to reflect that it is a specific term with meaning to the code.

**Comment by PI-Victor** [comment]
this enhancement is still labelled 'At Risk', no worries tho, once this clears, i will merge myself

/cancel approve

**Comment by PI-Victor** [comment]
ugh...

/approve cancel

**Comment by alculquicondor** [comment]
Note that the exception request was approved: https://groups.google.com/g/kubernetes-sig-release/c/2Zv7SZf-urE/m/jbAQ_zo8AAAJ

File: content/en/docs/reference/labels-annotations-taints.md [hunk 7]
```diff
@@ -268,6 +268,14 @@ Used on: Endpoints
 
 In Kubernetes clusters v1.21 (or later), the Endpoints controller adds this annotation to an Endpoints resource if it has more than 1000 endpoints. The annotation indicates that the Endpoints resource is over capacity.
 
+## scheduler.alpha.kubernetes.io/preferAvoidPods (deprecated)
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints.md:271]: I prefer to not make “deprecated” part of the fragment identifier.
```suggestion
## scheduler.alpha.kubernetes.io/preferAvoidPods (deprecated) {#scheduleralphakubernetesio-preferavoidpods}
```

File: content/en/docs/reference/scheduling/config.md [hunk 8]
```diff
@@ -58,6 +59,9 @@ extension points:
    Policy and are used to filter out nodes that can not run the Pod. Filters
    are called in the configured order. A pod is marked as unschedulable if no
    nodes pass all the filters.
+1. `PostFilter`: These plugins are called in their configured order when no
+   feasible nodes were found for the pod. If any postFilter plugin marks the Pod
```
Comment (sftim) [inline content/en/docs/reference/scheduling/config.md:63]: Is it PostFilter or postFilter?

File: content/en/docs/reference/labels-annotations-taints.md (hunk 7 above)
Comment (alculquicondor) [reply content/en/docs/reference/labels-annotations-taints.md]: Done

File: content/en/docs/reference/scheduling/config.md (hunk 8 above)
Comment (alculquicondor) [reply content/en/docs/reference/scheduling/config.md]: `postFilter` in the context of the configuration file. `PostFilter` in the context of the code interface that plugins implement. So, for the purpose of this doc, it should be `postFilter`. Fixed all the other references.

File: content/en/docs/reference/scheduling/config.md (hunk 5 above)
Comment (alculquicondor) [reply content/en/docs/reference/scheduling/config.md]: Done

**Comment by alculquicondor** [comment]
/hold cancel
Given that the exception was approved

**Pull request merged**